src/syncerd/types.rs

```diff
@@ -209,46 +213,72 @@ impl From<Boolean> for bool {
     }
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
+#[derive(Clone, Debug, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
 #[cfg_attr(
     feature = "serde",
     derive(Serialize, Deserialize),
     serde(crate = "serde_crate")
 )]
-#[display(Debug)]
 pub struct WatchTransaction {
     pub id: TaskId,
     pub lifetime: u64,
-    #[serde(with = "hex")]
-    pub hash: Vec<u8>,
+    pub hash: Txid,
     pub confirmation_bound: u32,
 }
 
-#[derive(Clone, Debug, Display, StrictEncode, StrictDecode, Eq, PartialEq, Hash)]
+impl fmt::Display for WatchTransaction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "WatchTransaction(id: {}, lifetime: {}, hash: {}, confirmations_bound: {})",
```

Comment:
Should be able to use `Display` derive here

Reply:
d1c3b39312a24e46f2fa8acd7df707ef454331e2